src/cluster/client/etcd/config.go

```diff
@@ -61,6 +61,11 @@ func (c ClusterConfig) NewCluster() Cluster {
 
 	if c.AutoSyncInterval > 0 {
 		cluster = cluster.SetAutoSyncInterval(c.AutoSyncInterval)
+	} else if c.AutoSyncInterval < 0 {
+		// Autosync should *always* be on, unless the user very explicitly requests it to be off, via a negative value
```

Comment:
Maybe put a comment in the `ClusterConfig` struct, otherwise someone explicitly putting in `autoSyncInterval: 0` might be confused as to why it doesn't work.